jerry-core/lit/lit-strings.c

```diff
@@ -282,6 +282,43 @@ lit_utf8_string_length (const lit_utf8_byte_t *utf8_buf_p, /**< utf-8 string */
 } /* lit_utf8_string_length */
 
 /**
+ * Calculate the required size of an utf-8 encoded string from cesu-8 encoded string
+ *
+ * @return size of an utf-8 encoded string
+ */
+lit_utf8_size_t
+lit_get_utf8_size_of_cesu8_string (const lit_utf8_byte_t *utf8_buf_p, /**< cesu-8 string */
+                                   lit_utf8_size_t utf8_buf_size) /**< string size */
+{
+  lit_utf8_size_t offset = 0;
+  lit_utf8_size_t reduction_size = 0;
```

Comment:
What about creating a variable here:

    lit_utf8_size_t utf8_buf_size = cesu8_buf_size;

And simply update its value:

    utf8_buf_size -= 2;